compiler/rustc_codegen_ssa/src/mir/statement.rs

```diff
@@ -115,6 +115,28 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 self.codegen_coverage(&mut bx, coverage.clone());
                 bx
             }
+            mir::StatementKind::CopyNonOverlapping(box mir::CopyNonOverlapping {
+                ref src,
+                ref dst,
+                ref count,
+            }) => {
+                let dst_val = self.codegen_operand(&mut bx, dst);
+                let src_val = self.codegen_operand(&mut bx, src);
+                let count_val = self.codegen_operand(&mut bx, count);
+                let count = count_val.immediate_or_packed_pair(&mut bx);
+                let dst = dst_val.immediate_or_packed_pair(&mut bx);
+                let src = src_val.immediate_or_packed_pair(&mut bx);
+                let flags = crate::MemFlags::empty();
+                bx.memcpy(
+                    dst,
+                    dst_val.layout.layout.align.pref,
+                    src,
+                    src_val.layout.layout.align.pref,
+                    count,
```

Comment:
This should be the number of bytes to copy, rather than the number of elements. Also the alignment used appears to be that of a pointer, rather than a pointee.

Reply:
I think I made this into # number of elements because cranelift expected # number of elements. Would it be better to convert this into # number of elements and divide by size in cranelift? I am unsure how to multiply `count` by pointee size as all I have here is an OperandRef::Value and no way to convert it into any sort of number